tests/unit/sagemaker/automl/test_auto_ml.py

```diff
@@ -348,6 +348,7 @@ def test_auto_ml_additional_optional_params(sagemaker_session):
         "generate_candidate_definitions_only": GENERATE_CANDIDATE_DEFINITIONS_ONLY,
         "tags": TAGS,
     }
+    assert "logs will be set to False. logs is only meaningful when wait is True." in caplog.text
```

Comment:
in addition to testing for the log output, you could `patch("sagemaker.automl.AutoMLJob.start_new")` and then call `.wait.assert_not_called()` on the resulting mock

Reply:
added a new test instead to test warning messages and `assert_not_called()`